resources.py

```diff
@@ -1057,15 +1057,29 @@ def network(self):
 
 class Volume(CloudscaleResource):
 
-    def __init__(self, request, api, size, zone, volume_type='ssd'):
+    def __init__(
+        self,
+        request,
+        api,
+        size=None,
+        zone=None,
+        volume_type='ssd',
+        volume_snapshot_uuid=None,
+    ):
         super().__init__(request, api)
 
-        self.spec = {
-            'name': f'{RESOURCE_NAME_PREFIX}-{uuid4().hex[:8]}',
-            'size_gb': size,
-            'type': volume_type,
-            'zone': zone
-        }
+        if volume_snapshot_uuid:
+            self.spec = {
+                'name': f'{RESOURCE_NAME_PREFIX}-{uuid4().hex[:8]}',
```

Comment:
Minor: template name to a variable and use in both cases to not duplicate the template